internal/gapicgen/cmd/genbot/main.go

```diff
@@ -114,17 +114,26 @@ func main() {
 	log.Println("checking if a pull request was already opened and merged today")
 	if pr, err := githubClient.GetRegenPR(ctx, "go-genproto", "closed"); err != nil {
 		log.Fatal(err)
-	} else if pr != nil {
-		now := time.Now()
-		today := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Local().Location())
-		log.Printf("Times -- Now: %v\tToday: %v\tPR: %v", now, today, pr)
-		if pr.Created.After(today) {
-			log.Println("skipping generation, already ran today")
-			return
-		}
+	} else if pr != nil && hasCreatedPRToday(pr.Created) {
+		log.Println("skipping generation, already created and merged a go-genproto PR today")
+		return
+	}
+	if pr, err := githubClient.GetRegenPR(ctx, "google-cloud-go", "closed"); err != nil {
+		log.Fatal(err)
+	} else if pr != nil && hasCreatedPRToday(pr.Created) {
+		log.Println("skipping generation, already created and merged a google-cloud-go PR today")
+		return
 	}
 
 	if err := generate(ctx, githubClient); err != nil {
 		log.Fatal(err)
 	}
 }
+
+// hasCreatedPRToday checks if the created time of a PR is from today.
+func hasCreatedPRToday(created time.Time) bool {
+	now := time.Now()
+	today := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Local().Location())
```

Comment:
this seems dubious. does the time zone match how Kokoro triggers the builds?

Reply:
This should just be time.UTC, would be more clear.

Reply:
Actually I re-thought about this I think now.Location() should be the right thing. This will be the timezone that kokoro uses

Reply:
Is it the same time zone that the kokoro scheduler uses, though?

Reply:
Or is this job run very frequently?

Reply:
The timezone will be that of the kokoro runner. This job is currently being run hourly.

Reply:
👍